src/managers/resources/adc.rs

```diff
@@ -0,0 +1,114 @@
+extern crate rppal;
+
+use crate::managers::Manager;
+use crate::managers::ManagerChannelData;
+use crate::managers::Resource;
+use crate::managers::ResourceManager;
+use crate::managers::TASK_SUCCESS;
+use crate::managers::macros::parse_channel_data;
+use crate::request::TaskData::EmgData;
+use crate::sgcp;
+use crate::sgcp::emg::*;
+use crate::todo;
+use anyhow::Error;
+use anyhow::Result;
+use anyhow::anyhow;
+use log::*;
+use rppal::gpio::{Gpio, OutputPin};
+use rppal::spi::{Bus, Mode, SlaveSelect, Spi};
+use std::process::Output;
+use std::{error::Error as StdError, io, io::Write, thread, time::Duration};
+
+
+use tokio::sync::mpsc::channel;
+use tokio::sync::mpsc::Receiver;
+use tokio::sync::mpsc::Sender;
+
+const DEFAULT_BUFFER_SIZE: usize = 100;
+const SPI_DEVICE_PATH: &str = "/dev/spidev0.0";
+const PAUSE_DURATION_MS: u64 = 500; // Pause duration in milliseconds
+
+pub fn process_data(values: Vec<u16>, inner_threshold: u16, outer_threshold: u16) -> Result<i32, Box<dyn StdError>> {
```

Comment:
Let's use `anyhow` errors and result types rather than the std library ones for consistency with the rest of the codebase